elide-core/src/main/java/com/yahoo/elide/graphql/Entity.java

```diff
@@ -0,0 +1,176 @@
+                Set<Entity> entitySet = new HashSet<>();
+                Class<?> loadClass = dictionary.getParameterizedType(this.entityClass, entry.getKey());
+                Boolean isToOne = dictionary.getRelationshipType(this.entityClass, entry.getKey()).isToOne();
+                if(isToOne) {
+                    entitySet.add(new Entity(Optional.of(this), ((Map<String, Object>) entry.getValue()), loadClass, this.requestScope));
+                } else {
+                    for(Map<String, Object> row : (List<Map<String, Object>>) entry.getValue()) {
+                        entitySet.add(new Entity(Optional.of(this), row, loadClass, this.requestScope));
+                    }
+                }
+                this.relationships.add(new Relationship(entry.getKey(), entitySet));
+            }
+        }
+    }
+
+    /**
+     * Get the relationship with name {@param name}
+     * @param name Name of relationship
+     * @return Relationship
+     */
+    public Optional<Relationship> getRelationship(String name) {
+        Iterator<Relationship> it = this.relationships.iterator();
+        while(it.hasNext()) {
+            Relationship relationship = it.next();
+            if(relationship.getName().equals(name)) {
+                return Optional.of(relationship);
+            }
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Get an Attribute with name {@param name}
+     * @param name Name of attribute
+     * @return Attribute
+     */
+    public Optional<Attribute> getAttribute(String name) {
+        Iterator<Attribute> it = this.attributes.iterator();
+        while(it.hasNext()) {
+            Attribute attribute = it.next();
+            if(attribute.getName().equals(name)) {
+                return Optional.of(attribute);
+            }
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Get the id of the entity
+     * @return the optional id
+     */
+    public Optional<String> getId() {
+        EntityDictionary dictionary = this.requestScope.getDictionary();
+        String idFieldName = dictionary.getIdFieldName(this.entityClass);
+        return this.attributes.stream()
+                .filter(entry -> idFieldName.equalsIgnoreCase(entry.name))
+                .map(e -> (String) e.value)
+                .findFirst();
+    }
+
+    /**
+     * Set the id of the entity if it doesn't have one already
+     */
+    public void setId() {
+        EntityDictionary dictionary = this.requestScope.getDictionary();
+        String idFieldName = dictionary.getIdFieldName(this.entityClass);
+        if(!getId().isPresent()) {
+            String uuid = UUID.randomUUID().toString()
+                    .replaceAll("[^0-9]", "")
+                    .substring(0, 3); //limit the number of digits to prevent InvalidValueException in PersistentResource.createObject()
+            //TODO: this is hacky, ask for a workaround for this.
+            this.data.put(idFieldName, uuid);
```

Comment:
Can we just remove it then?  data is private to Entity (or at least can be made so), and I don't think we need to fetch the ID from the data once it is set.